### ISSUE TITLE
SCDoc: Search.html: delayed typing feedback while search in progress

### ISSUE THREAD

**Issue by snappizz** [supercollider/supercollider#2253]
While typing in Search.html, the search field doesn't immediately show what I typed while another search is in progress.

This is a nuisance because I start by typing a single character such as "A". Thousands of entries match "A", so it takes a few seconds for it to compile those results. Meanwhile, if I continue to type into the field, the feedback on character entry is delayed until that search completes.

Last year I attacked this and managed to fix this problem and significantly speed up the search. Unfortunately I made some Git mistake and somehow lost my work. This is my reminder to restart.

**Comment by snappizz** [comment]
In case I suffer a tragic accident before I ever get around to this, I should leave some notes about what I did the first time around.

Instead of using a "keyup" event, Search does something pretty silly -- it polls the search field every 250ms with setInterval. I'm not sure why that would cause the field to lock up, but when I changed it to keyup the problem went away.

There are a few factors contributing to the slowness of the search, but the big one is the unnecessary use of regular expressions. As it turns out, regex is far slower than indexOf. IIRC there was also some inefficient code in displaying the search results, but I don't remember what.
